### ISSUE TITLE
fineuploader 5.1.3 throws an error when visiting pages with phantomjs 2.0

### ISSUE THREAD

**Issue by fschwahn** [FineUploader/fine-uploader#1363]
Hi,
we are using [poltergeist](https://github.com/teampoltergeist/poltergeist) for testing the more dynamic parts of our application, such as file uploads. poltergeist uses phantomjs as a test driver.

Since fineuploader 5.1.3 I'm seeing errors when phantomjs visits a page were fineuploader is used:

    TypeError: 'undefined' is not an object (evaluating 'new handlerType[handlerModuleSubtype + "UploadHandler"]')

This is on line 4525 of the unminified 5.1.3 build. It didn't happen on fineuploader 5.0.x . Do you have any idea why this is happening?

All the best,
Fabian

**Comment by rnicholus** [comment]
I'll need more information to track down that specific line (if it differs from the block of code pasted into your question).  There are many different types of combined fine uploader JS files.  

The error message is specific to Safari/webkit, which is what PhantomJS is based off of.  This error suggests that the proper uploader namespace does not exist when you are running your tests.  For example, if you are trying to test Fine Uploader S3 or Azure, the `qq.s3` or `qq.azure` namespaces are missing. Or if you are testing traditional Fine Uploader, the `qq.traditional` namespace is missing.  

You haven't provided any detail about your tests or even the type of build you are trying to test, so I can't give you any conclusive answers.  You'll need to look closer at the file or files  you are making available to your tests, and perhaps how Fine Uploader is being initialized.

**Comment by fschwahn** [comment]
First of all, the only thing I changed was fineuploader, everything else is identical. Were there any breaking changes between 5.0 and 5.1? I couldn't find anything on the blog, but maybe I missed somewhere else?

As for the line, in my version of fineuploader the string `handlerModuleSubtype + "UploadHandler"` just comes up once, in `upload.initHandler` in `qq.UploadHandlerController`.

My setup is that I do a direct upload to s3 in development/production, and traditional upload in my test environment for integration testing. This worked in the past without a problem. I just switched my development environment to the traditional uploader, and now I'm seeing the same error in Chrome.

So the traditional uploader does not work for me anymore in 5.1, while the same setup worked in 5.0. Maybe I was always doing something wrong, but came by in the past? I'm using the jQuery-plugin, maybe it is connected to that.

**Comment by rnicholus** [comment]
No, there were no breaking changes, unintentional or otherwise.  Breaking changes are always represented as a new major version (i.e. 5.0 vs 6.0).  

If you are having the same issue with the traditional uploader outside of your unit tests, then there is definitely an issue with your code or environment, because all uploaders are working as expected.  In fact, we have our own (large) suite of unit tests that hit S3, Azure, and the traditional uploader. 

If this is reproducible in Chrome simply on page load, please post all relevant code needed to reproduce your issue (markup and JS), and we will take a look.

**Comment by fschwahn** [comment]
In my effort to reproduce the problem outside my application I downloaded fineuploader again from the homepage and when I saw the download page I realized that the traditional uploader is not part of the s3 package I downloaded. I just assumed that the traditional uploader is always part of the package. In earlier versions I just used the builder to include both.

I realize that this was entirely my fault, but a message at least in debug mode if you are trying to use an endpoint which is not included would go a long way.

Sorry for wasting your time.

All the best,
Fabian
